### ISSUE TITLE
HTTPS config has weaknesses that could allow for attack

### ISSUE THREAD

**Issue by nzkarit** [jlund/streisand#279]
Reproduction:
1. Install Streisand as per instructions
2. Set up a DNS entry for the server (as SSL Labs won't check an IP, only a domain on 443) (Can use https://testssl.sh/ to check IPs and on non 443)
3. Go to SSL Labs (https://www.ssllabs.com/ssltest/index.html) to test the configuration (click through the warning about self signed and that there is a mismatch (cert for IP and you accessing the domain name))
4. Read the output

Issues:
1. SSLv3 enabled and SSLv3 has known weaknesses. Even though nothing by default connects to SSLv3 allows for downgrade attacks. Recommendation is to disable SSLv3.
2. Uses common DH Primes which can be in the logjam attack (https://weakdh.org/). Recommendation set up a step to regenerate the DH Primes preferably with a length of 2048 or greater (https://weakdh.org/sysadmin.html)
3. HSTS is no enabled. HSTS (https://www.owasp.org/index.php/HTTP_Strict_Transport_Security) is a HTTP Header that will tell the browser to never access the site over HTTP. This will stop SSL Stripping attacks on non first visit to the site. There is also the HSTS preload list which will protect against the first visit to the site. https://hstspreload.appspot.com/ https://scotthelme.co.uk/hsts-preloading/

Mozilla provide a config generator which will get an A+ on SSL Lab https://mozilla.github.io/server-side-tls/ssl-config-generator/

**Comment by jlund** [comment]
1. SSLv3 is now explicitly disabled. However, I hope that no one is still using a browser from mid-2014 that supports the protocol. It was entirely removed from every mainstream client a long time ago. If people are still on an affected version, they've got nearly two years of patches that they have failed to apply. SSLv3 is the least of their problems :)
2. Fixed. This was already happening for OpenVPN. I made it more generic.
3. HSTS has [been enabled](https://github.com/jlund/streisand/blob/master/playbooks/roles/streisand-gateway/templates/vhost.j2#L7) since day one. What's interesting is that the SSL Labs check doesn't receive the header because it gets blocked by HTTP basic authentication. The header is being sent to all users after a successful login though.

We won't be able to do any automatic HSTS preloading for random IP addresses that typically don't have an assigned hostname, but I hope that HTTP continues to die.

I used Mozilla's generator for the new Nginx vhost configuration. Some of the hot and new-ish cipher suites (like ChaCha20) won't work on Ubuntu 14.04, but they'll come along for the ride when 16.04 becomes the new foundation.

Thanks for the report!
